src/demangle.cpp

```diff
@@ -25,80 +25,23 @@ char* Demangle::demangleCpp(const char* s) {
     return result;
 }
 
-char* Demangle::demangleRust(const char* s, const char* e) {
+char* Demangle::demangleRust(const char *s, struct demangle const *demangle, bool full_signature) {
```

Comment:
Seems like `s` argument is redundant here: it is only used for initial size estimation. `struct demangle` already has everything needed to estimate result size even better.

Reply:
I think that is smarter